autobahn/websocket/types.py

```diff
@@ -127,6 +127,96 @@ def __str__(self):
         return json.dumps(self.__json__())
 
 
+@public
+class ConnectingRequest(object):
+    """
+    Thin-wrapper for WebSocket connection request information provided in
+    :meth:`autobahn.websocket.protocol.WebSocketClientProtocol.onConnecting`
+    after a client has connected, but before the handshake has
+    proceeded.
+
+    `host`, `port`, and `resource` are all required, everything else
+    is optional. Note that these are values that will be seen by the
+    client and should represent the public-facing host, port and
+    resource to which the client is connecting (not necessarily the
+    action host/port being used).
+    """
+
+    __slots__ = (
+        'host',
+        'port',
+        'resource',
+        'headers',
+        'useragent',
+        'origin',
+        'protocols',
+    )
+
+    def __init__(self, host, port, resource, headers=None, useragent=None, origin=None, protocols=None):
+        """
+        :param headers: HTTP headers to send in the opening handshake
+        :type headers: dict
+
+        """
+        # required
+        self.host = host
+        self.port = port
+        self.resource = resource
+        # optional
+        self.headers = headers if headers else dict()
+        self.useragent = useragent
+        self.origin = origin
+        self.protocols = protocols if protocols else []
+
+    def __json__(self):
+        return {
+            'host': self.host,
+            'port': self.port,
+            'resource': self.resource,
+            'headers': self.headers,
+            'useragent': self.useragent,
+            'origin': self.origin,
+            'protocols': self.protocols,
+        }
+
+    def __str__(self):
+        return json.dumps(self.__json__())
+
+
+@public
+class TransportDetails(object):
+    """
+    Details of our transport made available to the `onConnecting`
+    callback.
+    """
+
+    __slots__ = (
+        'peer',  # .getPeer() on Twisted, .get_extra_info('peername') on asyncio
+        'host',  # .getHost() on Twisetd, .get_extra_info('sockname') on asyncio
+    )
+    # possibly useful:
+    # is_secure
+    # peer_certificate  # getPeerCertificate(), .get_extra_info('peercert')
+
+    def __init__(self, peer, host):
```

Comment:
Similar to my above comment: what about UDS, pipes, serial, ..?

Reply:
I believe they all return "something" on their respective platforms (i.e. `ITransport` in Twisted has `getHost()` and it returns an `IAddress` which is .. basically "transport specific" (e.g. TCPAddress or UNIXAddress, etc). That said, I'll double-check pipes though.